internal/protobuf/helpers.go

```diff
@@ -62,3 +64,14 @@ func AsMessages(messages interface{}) []proto.Message {
 
 	return protos
 }
+
+// MustMarshalAny marshals a protobug into an any.Any type, panicing
+// if that operation fails.
+func MustMarshalAny(pb proto.Message) *any.Any {
+	a, err := ptypes.MarshalAny(pb)
+	if err != nil {
+		panic(err.Error())
```

Comment:
Contour will crash if this fails? Should we return the error and let the caller decide what action to take?

Reply:
I agree that in general we shouldn't panic, but here, if the function returns an error as well, it can't be used as an inline helper, and will require caller-location error handling. So you can't just use it inline in structs any more to help with brevity.

It's actually the same behaviour as the thing it's replacing. I think it's probably okay in this case because we control the generation of the objects that are being marshalled; if anything does go wrong, things are in a bad enough state that Contour should stop working.

Reply:
Yup, there is no change in behaviour, but the `Must*` naming convention makes it clearer at the callsite that it might panic.

Previously on slack, I've argued that the way Contour uses this helper is safe because we never directly marshall anything the user gives us. If we internally generate a protobuf marshaling error, we are very broken :)